dashboard/app/models/user.rb

```diff
@@ -555,6 +555,8 @@ def memoized_teachers
 
   before_save :remove_cleartext_emails, if: -> {student? && migrated? && user_type_changed?}
 
+  before_save :remove_family_name, if: -> {DCDO.get('family-name-features', false) && sections_as_pl_participant.any?}
```

Comment:
Would it make more sense to use `validate` here? Essentially validating that teachers don't have a family name?